docs/aws/audit/kmsmonitoring/rules/secret_encrypted_kms_cmk.mdx

```diff
@@ -23,6 +23,281 @@ ISO27001, HIPAA, NISTCSF, PCIDSS
+       KmsKeyId=cmk_id
+   )
+   print(f"Created secret with ARN: {response['ARN']}")
+   ```
+
+### Azure (Microsoft Azure)
+
+1. **Install Azure SDK for Python:**
+   ```bash
+   pip install azure-identity azure-keyvault-secrets azure-keyvault-keys
+   ```
+
+2. **Create a Key Vault and a Key:**
+   ```python
+   from azure.identity import DefaultAzureCredential
+   from azure.keyvault.keys import KeyClient
+   from azure.keyvault.secrets import SecretClient
+
+   credential = DefaultAzureCredential()
+   key_vault_name = "myKeyVault"
+   key_vault_uri = f"https://{key_vault_name}.vault.azure.net"
+
+   key_client = KeyClient(vault_url=key_vault_uri, credential=credential)
+   key_name = "myCMK"
+   key = key_client.create_key(key_name, "RSA")
+   print(f"Created key with name: {key.name}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   secret_client = SecretClient(vault_url=key_vault_uri, credential=credential)
+   secret_name = "mySecret"
+   secret_value = "mySecretValue"
+
+   secret = secret_client.set_secret(secret_name, secret_value, key_id=key.id)
+   print(f"Created secret with name: {secret.name}")
+   ```
+
+### GCP (Google Cloud Platform)
+
+1. **Install Google Cloud SDK for Python:**
+   ```bash
+   pip install google-cloud-secret-manager google-cloud-kms
+   ```
+
+2. **Create a KMS Key:**
+   ```python
+   from google.cloud import kms_v1
+
+   client = kms_v1.KeyManagementServiceClient()
+   project_id = "my-project"
+   location_id = "global"
+   key_ring_id = "my-key-ring"
+   key_id = "my-key"
+
+   key_ring_name = client.key_ring_path(project_id, location_id, key_ring_id)
+   key_ring = client.create_key_ring(request={"parent": key_ring_name, "key_ring_id": key_ring_id})
+
+   key_name = client.crypto_key_path(project_id, location_id, key_ring_id, key_id)
+   key = client.create_crypto_key(request={"parent": key_ring_name, "crypto_key_id": key_id, "crypto_key": {"purpose": kms_v1.CryptoKey.CryptoKeyPurpose.ENCRYPT_DECRYPT}})
+   print(f"Created key with name: {key.name}")
+   ```
+
+3. **Create a Secret with CMK:**
+   ```python
+   from google.cloud import secretmanager
+
+   client = secretmanager.SecretManagerServiceClient()
+   project_id = "my-project"
+   secret_id = "my-secret"
+   secret_value = "my-secret-value"
+
+   parent = f"projects/{project_id}"
+   secret = client.create_secret(
+       request={
+           "parent": parent,
+           "secret_id": secret_id,
+           "secret": {"replication": {"automatic": {}}},
+           "encryption": {"kms_key_name": key.name}
+       }
+   )
+
+   payload = secretmanager.SecretPayload(data=secret_value.encode("UTF-8"))
+   client.add_secret_version(request={"parent": secret.name, "payload": payload})
+   print(f"Created secret with name: {secret.name}")
+   ```
+
+These steps will help you ensure that your secrets are encrypted with Customer Managed Keys (CMKs) in AWS, Azure, and GCP using Python scripts.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>